### PR TITLE
EIP standard deposit and withdraw functions

### PR DESCRIPTION
**Pull request by gabririgo**
I am proposing a new EIP to establish a standard interface for deposit and withdraw functions used in many contracts. They are abstracted functions which allow for ETH and ERC20 movements, allow external applications to pre-check for transaction success.
The new EIP has been created using the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-X.md

### CODE REVIEW

**Comment by MicahZoltu** [comment]
Is this intended to work with ERC20 tokens?  If so, is the expectation that the caller first approves the target contract and then the target contract withdraws?

Can you provide some example scenarios for when this would be valuable for cross-contract interactions?  Most of the scenarios I can think of don't involve contracts depositing/withdrawing to other contracts, and when it is contracts talking to other contracts they tend to not be arbitrary, the contracts are tightly integrated with each other.

**Comment by escursionista** [comment]
this is intended to work with Ether deposits and withdraws as well, and also Ether as a token if in the future Ether itself becomes an ERC20 token. 
most decentralized exchanges use a deposit/withdraw function and the design of a generalized interface for those functions would lead to higher compatibility.
For ERC20 tokens, there is the expectation for approval. examples of deposits/withdrawals are decentralized exchanges (0x, idex just as an example, but there are many more who use such functions).
standard ERC20 exchanges can work without deposit/withdraw functions, but most of the exchanges use those functions. The interface I proposed is generalized in order to be applicable by all of them.

**Comment by MicahZoltu** [comment]
Can you describe a tool that would benefit from standardization?  Currently, most exchanges provide their own dApp UI that integrates with the exchange.  In what situation would a user be interacting with an exchange without using the exchanges UI?

**Comment by escursionista** [comment]
a user could deposit from another dapp and then trade on a decentralized exchange UI. fully decentralized exchanges can be used by other applications without the need for an exchange ui. Also, dapps could be embedded in the exchange ui.

**Comment by MicahZoltu** [comment]
I'm struggling to understand the utility of dApps being able to deposit into arbitrary other apps.  Why would dApp A need a function to deposit into _arbitrary_ dApp B?  If dApp A is integrated with dApp B directly, then it can follow whatever pattern dApp B supports (no need for a standard).  A standard is only necessary if a dApp wants the ability to deposit into arbitrary other dApps without knowing about their existence ahead of time.

In the case of exchanges, if I write a dApp that integrates with EtherDelta and 0x, the two don't need to share an interface (and in fact, both have a different pattern for "deposits" already).  I only need a shared interface if I want to write a dApp that can deposit into any exchange that exists now or may exist in the future without me needing to know about it or integrate it, the user would just provide the exchange address and be done.

File: EIPS/eip-draft_sandardize_abbrev.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: <EIP standard deposit and withdraw functions>
```
Comment (pirapira) [inline EIPS/eip-draft_sandardize_abbrev.md]: The angle brackets look odd.

File: EIPS/eip-draft_sandardize_abbrev.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    Author: <Gabriele Rigo, gab@rigoblock.com>
```
Comment (pirapira) [inline EIPS/eip-draft_sandardize_abbrev.md]: The angle brackets should enclose the email address only.

File: EIPS/eip-draft_sandardize_abbrev.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    Type: <Standard Track>
```
Comment (pirapira) [inline EIPS/eip-draft_sandardize_abbrev.md]: Angle brackets are not needed here and on the next two lines.

**Comment by Arachnid** [comment]
This is a courtesy notice to let you know that the format for EIPs has been modified slightly. If you want your draft merged, you will need to make some small changes to how your EIP is formatted:

 - Frontmatter is now contained between lines with only a triple dash ('---')
 - Headers in the frontmatter are now lowercase.

If your PR is editing an existing EIP rather than creating a new one, this has already been done for you, and you need only rebase your PR.

In addition, a continuous build has been setup, which will check your PR against the rules for EIP formatting automatically once you update your PR. This build ensures all required headers are present, as well as performing a number of other checks.

Please rebase your PR against the latest master, and edit your PR to use the above format for frontmatter. For convenience, here's a sample header you can copy and adapt:

```
---
eip: <num>
title: <title>
author: <author>
type: [Standards Track|Informational|Meta]
category: [Core|Networking|Interface|ERC] (for type: Standards Track only)
status: Draft
created: <date>
---
```

**Comment by axic** [comment]
I suggest to close this PR since the author hasn't responded to editor's requests since 2017.

@gabririgo do you still want to pursue this?

**Comment by gabririgo** [comment]
I agree, closing the issue. The dex standardization has been successfully implemented at layer2 protocols like 0x. Furthermore, dapps have abstracted the way they interact with the different DEXes in order to accommodate for different deposit/withdraw interfaces.

**Pull request closed without merging**